### PR TITLE
Remove theme + init subcommands

### LINKED ISSUE #111: Remove themes
`bsb -init` and `-theme` are bloated and very hard to maintain. 

I'd prefer if we could maintain a set of blessed project templates in the melange-re org that folks can just clone with git. Or otherwise use [spin](https://github.com/tmattio/spin).

- 👍 It should be easier to find them
- 👍 it should be easier to contribute / keep them up to date with new releases of melange
- 👍 we don't bundle a virtual filesystem inside the melange binary

### PR DESCRIPTION
**Pull request by jchavarri**
Fixes #111.

I remove some modules as well that were only used for init or theme, e.g. `bsb_regex`.

To compensate for this, I've created a super simple template in https://github.com/jchavarri/melange-basic-template. Happy to move it to `melange-re` org if it makes sense :)

### CODE REVIEW

**Comment by EduardoRFS** [comment]
Thankfully this was really isolated, nice job and welcome as a contributor.

anime girl impressed with -4k lines of code

**Review by EduardoRFS — APPROVED** [review]

**Review by anmonteiro — APPROVED** [review]
This is great, and I'm all for it.

What do folks think about removing the CLI options for `-theme`, `-themes` and `-init`, thought?

**Comment by EduardoRFS** [comment]
@anmonteiro I think if someone is relying on this it would be better to remove the CLI and let it fail than just ignoring it.

But if someone is using this please let us know

**Comment by anmonteiro** [comment]
I wasn't suggesting we ignore it, but rather point them to our project template

**Comment by jchavarri** [comment]
> point them to our project template

This makes sense, I'll add some message for `-init`, `-theme` and `-themes` to point to the template.

**Comment by jchavarri** [comment]
Added back the flags with a notice in 2d39f23:

```bash
$ esy x bsb -help
The subcommands -init, -theme and -themes are deprecated now. Use the template at https://github.com/melange-re/melange-basic-template to start a new project.
```

File: jscomp/main/bsb_main.ml [hunk 1]
```diff
@@ -167,22 +169,16 @@ module CLI = struct
 
   let init_arg =
     let docv = "init path" in
-    let doc =
-      "Init sample project to get started. \n\
-       Note: (`bsb -init sample` will create a sample project while \n\
-       `bsb -init .` will reuse current directory)" in
+    let doc = "Currently does nothing. Enabled for compatibility." in
```
Comment (anmonteiro) [inline jscomp/main/bsb_main.ml:172]: could we also add the theme link in these docstrings?

Comment (jchavarri) [reply jscomp/main/bsb_main.ml]: done!

**Comment by bikallem** [comment]
Probably a dumb question. Just saw dune-project file in the sample template project, is melange project a dune project already?

**Comment by jchavarri** [comment]
Yes, it's integrated with dune, up to a point. That integration was (afaik) the main driver behind the idea of Melange that Antonio started :) The integration with dune is not full yet, dune handles low level rules that will call bsc and bsb-helper. But it's a big milestone to better integrate with the OCaml ecosystem.

**Pull request merged**